packages/databyss-ui/editor/EditorPage.js

```diff
@@ -25,29 +27,46 @@ import { isBlockEmpty, isEmptyAndAtomic } from './slate/slateUtils'
 const EditorPage = ({ children, autoFocus }) => {
   const [editorState, dispatchEditor] = useEditorContext()
   const { setSource } = useSourceContext()
+  const { setTopic } = useTopicContext()
 
-  const { sources, newSources, editableState } = editorState
+  const { sources, newSources, topics, newTopics, editableState } = editorState
 
   /*
   checks to see if new source has been added
   adds the new source to the source provider
   */
+
   useEffect(
     () => {
-      if (newSources && editableState) {
-        if (newSources.length > 0) {
-          newSources.forEach(s => {
-            const _source = {
-              _id: s._id,
-              text: { textValue: s.textValue, ranges: s.ranges },
-            }
-            setSource(_source)
-            dispatchEditor(removeSourceFromQueue(s._id))
-          })
+      if (editableState) {
+        if (newSources) {
```

Comment:
Combine these to `newAtomics` as discussed. 
Also, rather than nested `if`s, combine to `if (newAtomics && newAtomics.length > 0)`